src/main/java/edu/harvard/iq/dataverse/globus/GlobusServiceBean.java

```diff
@@ -0,0 +1,717 @@
+            status = task.getStatus();
+        }
+        if (result.status != 200) {
+            globusLogger.warning("Cannot find information for the task " + taskId + " : Reason :   " + result.jsonResponse.toString());
+        }
+
+        return task;
+    }
+
+    public Boolean getTaskSkippedErrors(AccessToken clientTokenUser, String taskId , Logger globusLogger) throws MalformedURLException {
+
+        URL url = new URL("https://transfer.api.globusonline.org/v0.10/endpoint_manager/task/"+taskId );
+
+        MakeRequestResponse result = makeRequest(url, "Bearer",clientTokenUser.getOtherTokens().get(0).getAccessToken(),
+                "GET",  null);
+
+        Task task = null;
+
+        if (result.status == 200) {
+            task = parseJson(result.jsonResponse, Task.class, false);
+            return task.getSkip_source_errors();
+        }
+
+        return false;
+    }
+
+    public AccessToken getClientToken() throws MalformedURLException {
+        String basicGlobusToken = settingsSvc.getValueForKey(SettingsServiceBean.Key.BasicGlobusToken, "");
+        URL url = new URL("https://auth.globus.org/v2/oauth2/token?scope=openid+email+profile+urn:globus:auth:scope:transfer.api.globus.org:all&grant_type=client_credentials");
+
+        MakeRequestResponse result = makeRequest(url, "Basic",
+                basicGlobusToken,"POST",   null);
+        AccessToken clientTokenUser = null;
+        if (result.status == 200) {
+            clientTokenUser = parseJson(result.jsonResponse, AccessToken.class, true);
+        }
+        return clientTokenUser;
+    }
+
+    public AccessToken getAccessToken(HttpServletRequest origRequest, String basicGlobusToken ) throws UnsupportedEncodingException, MalformedURLException {
+        String serverName = origRequest.getServerName();
+        if (serverName.equals("localhost")) {
+            serverName = "utl-192-123.library.utoronto.ca";
+        }
+
+        String redirectURL = "https://" + serverName + "/globus.xhtml";
+
+        redirectURL = URLEncoder.encode(redirectURL, "UTF-8");
+
+        URL url = new URL("https://auth.globus.org/v2/oauth2/token?code=" + code + "&redirect_uri=" + redirectURL
+                + "&grant_type=authorization_code");
+        logger.info(url.toString());
+
+        MakeRequestResponse result = makeRequest(url, "Basic", basicGlobusToken,"POST",   null);
+        AccessToken accessTokenUser = null;
+
+        if (result.status == 200) {
+            logger.info("Access Token: \n" + result.toString());
+            accessTokenUser = parseJson(result.jsonResponse, AccessToken.class, true);
+            logger.info(accessTokenUser.getAccessToken());
+        }
+
+        return accessTokenUser;
+
+    }
+
+
+    public MakeRequestResponse  makeRequest(URL url, String authType, String authCode, String method, String jsonString) {
+        String str = null;
+        HttpURLConnection connection = null;
```

Comment:
I'm not sure how important it is with Globus, but we use the Apache libraries elsewhere which can keep connections open and avoid the overheads of recreating connections.